decidim-accountability/app/queries/decidim/accountability/metrics/results_metric_manage.rb

```diff
@@ -26,10 +23,8 @@ def save
                                                            related_object_type: "Decidim::Component",
                                                            related_object_id: related_object_id)
             record.assign_attributes(cumulative: cumulative_value, quantity: quantity_value)
-            @registry << record
+            record.save!
           end
-          @registry.each(&:save!)
-          @registry
```

Comment:
Note that this is changing the API of the method (you were returning the list of records, now it's returning the list of `cumulative`), I don't know if this affects somewhere else down the line 😄

Reply:
Yes, it changes the returned value but it was used by tests (which are already refactored) but not by the background jobs, which rely on the persisted records. :+1: